### PR TITLE
E commerce

### PR DESCRIPTION
**Pull request by abrarqa62**
I did the fully automation of E-commerce page.

https://user-images.githubusercontent.com/106743757/173621843-f6a23a05-642f-4909-89a3-e9964e54810d.mp4

### CODE REVIEW

File: cypress/integration/EcommerceAutomation.spec.js [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+import { NavigateTo } from "../support/page_objects/PageObjectsEcommerce";
+
+describe("Ecommerce Page", () => {
+  before("Run the Website", () => {
+    cy.visit("/");
+  });
+
+  //    Profit card
+  context("When user move on profit card", () => {
+    it("User is on Profit Card Section", () => {
```
Comment (hassanatdevsinc) [inline cypress/integration/EcommerceAutomation.spec.js]: Text in 'it' block should start with "Verify if "

File: cypress/integration/EcommerceAutomation.spec.js [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+        let HeightCanvas = $canvas.height();xdescribe
+
+        let pointXaxis1 = WidthCanvas * 0.71;
+        let PointYaxis1 = HeightCanvas * 0.63;
+        let pointXaxis2 = WidthCanvas * 0.75;
+        let PointYaxis2 = HeightCanvas * 0.65;
+        let pointXaxis3 = WidthCanvas * 0.92;
+        let PointYaxis3 = HeightCanvas * 0.62;
+
+        c.click(pointXaxis1, PointYaxis1).wait(300);
```
Comment (hassanatdevsinc) [inline cypress/integration/EcommerceAutomation.spec.js]: please remove static waits from  test cases

File: cypress/integration/EcommerceAutomation.spec.js [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+        e.click(10, 130).wait(300);
+        e.click(40, 110).wait(300);
+        e.click(75, 120).wait(300);
+        e.click(105, 130).wait(300);
+
+      });
+    });
+
+      it('User now select Tether from Dropdown', () => {
+        NavigateTo.DropdownButtonEarning().click().should('be.visible');
```
Comment (hassanatdevsinc) [inline cypress/integration/EcommerceAutomation.spec.js]: NavigateTo.DropdownButtonEarning().click().should('be.visible'); in this line the assertion for be visible should be before click action

File: cypress/integration/EcommerceAutomation.spec.js [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+
+        });
+
+      });
+      it('User now select Ehtereum from Dropdown', () => {
+        NavigateTo.DropdownButtonEarning().click().should('be.visible');
+        NavigateTo.DropdownListEarning().contains('Ethereum').click();
+
+        NavigateTo.EarningChart().then(($Earningcanvas) => {
+          let e = cy.wrap($Earningcanvas);
```
Comment (hassanatdevsinc) [inline cypress/integration/EcommerceAutomation.spec.js]: let e = cy.wrap($Earningcanvas);
        cy.log($Earningcanvas.width());
        cy.log($Earningcanvas.height());
 As this is used in most of the test cases please create a function for it and use it in the test cases

File: cypress/integration/EcommerceAutomation.spec.js [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+  // Earning Card
+
+  context("When user now move to Earning Card", () => {
+    it("Test the Daily Income Chart", () => {
+      NavigateTo.EarningChart().then(($Earningcanvas) => {
+        let e = cy.wrap($Earningcanvas);
+        cy.log($Earningcanvas.width());
+        cy.log($Earningcanvas.height());
+
+        e.click(10, 130).wait(300);
```
Comment (hassanatdevsinc) [inline cypress/integration/EcommerceAutomation.spec.js]: why do hardcoding values in the click use variables in it.

File: cypress/support/page_objects/PageObjectsEcommerce.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  // Earning Card
+
+  EarningChart() {
+    return cy.get('[data-test-id="EarningChart"]');
+  }
+  DropdownButtonEarning() {
+    return cy.get('[data-test-id="EarningDropdownButton"]');
+  }
+  DropdownListEarning() {
+    return cy.get(".option-list");
```
Comment (hassanatdevsinc) [inline cypress/support/page_objects/PageObjectsEcommerce.js]: why not used Data-test-id here?

File: cypress/support/page_objects/PageObjectsEcommerce.js [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+  TrafficFrontDropdownButton() {
+    return cy.get('[data-test-id="FrontTrafficdropdown"]');
+  }
+
+  TrafficCardFront() {
+    return cy.get('[data-test-id="TrafficCardFront"]');
+  }
+  FrontTrafficDropdownList() {
+    return cy.get(".option-list");
```
Comment (hassanatdevsinc) [inline cypress/support/page_objects/PageObjectsEcommerce.js:53]: same as on line 31

**Review by hassanatdevsinc — CHANGES_REQUESTED** [review]
Please add the screenshots and screen recordings folder in git ignore.
I have added comments please look into them. Do let me know if you have any questions Thanks

**Review by hassanatdevsinc — APPROVED** [review]

File: cypress/support/page_objects/PageObjectsEcommerce.js (hunk 6 above)
Comment (abrarqa62) [reply cypress/support/page_objects/PageObjectsEcommerce.js]: Done

File: cypress/support/page_objects/PageObjectsEcommerce.js (hunk 7 above)
Comment (abrarqa62) [reply cypress/support/page_objects/PageObjectsEcommerce.js:53]: Done

File: cypress/integration/EcommerceAutomation.spec.js (hunk 3 above)
Comment (abrarqa62) [reply cypress/integration/EcommerceAutomation.spec.js]: Done

File: cypress/integration/EcommerceAutomation.spec.js (hunk 2 above)
Comment (abrarqa62) [reply cypress/integration/EcommerceAutomation.spec.js]: Done

**Pull request merged**